### PR TITLE
feat(Caching) : Support cache key formatting

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

feat(Caching) : Support cache key formatting

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md [hunk 1]
```diff
@@ -37,7 +37,10 @@ Install-Package Masa.Contrib.Caching.Distributed.StackExchangeRedis
 2. 添加Redis缓存
 
 ```C#
-builder.Services.AddStackExchangeRedisCache();
+builder.Services.AddDistributedCache(distributedCacheOptions =>
+{
+    distributedCacheOptions.UseStackExchangeRedisCache();
```
Comment (doddgu) [inline src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md:42]: How to use it with dcc?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/Extensions/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License. See LICENSE.txt in the project root for license information.
 
 // ReSharper disable once CheckNamespace
+
```
Comment (doddgu) [inline src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/Extensions/ServiceCollectionExtensions.cs:5]: Is resharper's rule blank or non-blank?

File: src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md [hunk 3]
```diff
@@ -77,6 +77,65 @@ string key = "test_1";
 distributedCacheClient.Set(key, "test_content");
 ```
 
+#### 用法3 (结合Dcc使用)：
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md:80]: There is a slight difference between the first step and the second step

[the remark this answers is not in the document]
Comment (doddgu) [reply src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md:80]: I mean your document doesn't make it clear

[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/Contrib/Caching/Distributed/Masa.Contrib.Caching.Distributed.StackExchangeRedis/README.zh-CN.md:80]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**